.github/workflows/main.yml

```diff
@@ -248,8 +248,47 @@ jobs:
       - name: Test
         run: pytest
 
+  test-python-macos-arm64:
+    runs-on: flyci-macos-large-latest-m1
+    needs: [build-cpp-test-macos, build-and-test-python]
+    if: (github.event_name == 'push') || (github.event_name == 'workflow_dispatch')
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      
+      - name: Check python version
+        run: |
+          uname -a
+          uname -m
+          python -c 'import platform; print(platform.platform())'
+          python --version
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: version
+          path: .
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: wheelhouse-macos
+          path: wheelhouse-macos/.
+   
+      - name: Install
+        run: |
+          pip install power-grid-model[dev]==$(cat PYPI_VERSION) --find-links=wheelhouse-macos
+      
+      - name: test
+        run: |
+          pytest
+
+
   publish-wheels:
-    needs: [build-cpp-test-linux, build-cpp-test-windows, build-cpp-test-macos, build-and-test-python, build-and-test-conda]
+    needs: [build-cpp-test-linux, build-cpp-test-windows, build-cpp-test-macos, build-and-test-python, build-and-test-conda, test-python-macos-arm64]
```

Comment:
publish wheels was skipped

Reply:
Use the solution here:

https://stackoverflow.com/questions/69354003/github-action-job-fire-when-previous-job-skipped